biodatasets/medquad/medquad.py

```diff
@@ -0,0 +1,264 @@
+N.B. We removed the answers from 3 subsets to respect the MedlinePlus copyright (https://medlineplus.gov/copyright.html):
+(1) A.D.A.M. Medical Encyclopedia, (2) MedlinePlus Drug information, and (3) MedlinePlus Herbal medicine and supplement information.
+-- We kept all the other information including the URLs in case you want to crawl the answers. Please contact me if you have any questions.
+
+For more info please visit https://github.com/abachaa/MedQuAD/
+"""
+import json
+import os
+import xml.etree.ElementTree as ET
+from typing import List, Tuple, Dict
+
+import datasets
+import requests
+
+from utils import schemas
+from utils.configs import BigBioConfig
+from utils.constants import Tasks
+
+_CITATION = """\
+@article{BenAbacha-BMC-2019,
+  author    = {Asma {Ben Abacha} and Dina Demner{-}Fushman},
+  title     = {A Question-Entailment Approach to Question Answering},
+  journal   = {{BMC} Bioinform.},
+  volume    = {20},
+  pages     = {511:1--511:23},
+  year      = {2019},
+  url       = {https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3119-4},
+  doi       = {https://doi.org/10.1186/s12859-019-3119-4},
+  biburl    = {https://citation-needed.springer.com/v2/references/10.1186/s12859-019-3119-4?format=refman&flavour=citation},
+}
+"""
+
+_DATASETNAME = "medquad"
+
+_DESCRIPTION = """\
+MedQuAD: Medical Question Answering Dataset  
+MedQuAD includes 47,457 medical question-answer pairs created from 12 NIH websites\
+(e.g. cancer.gov, niddk.nih.gov, GARD, MedlinePlus Health Topics).\
+The collection covers 37 question types (e.g. Treatment, Diagnosis, Side Effects) associated with diseases,\
+drugs and other medical entities such as tests.
+"""
+
+_HOMEPAGE = "https://github.com/abachaa/MedQuAD"
+
+_LICENSE = "https://creativecommons.org/licenses/by/4.0/legalcode"  # TODO: terms aren't available in the repository! In the issue it is 'CC BY 4.0'
+
+_DATA_PATH = "https://raw.githubusercontent.com/abachaa/MedQuAD/master"
+
+_DATA_REPO_FETCH_URL = "https://api.github.com/repos/abachaa/MedQuAD/git/trees/master?recursive=1"
+
+_SUBSET_BASE_URIS = {
+    "cancergov_qa": "1_CancerGov_QA",
+    "gard_qa": "2_GARD_QA",
+    "ghr_qa": "3_GHR_QA",
+    "mplus_health_topics_qa": "4_MPlus_Health_Topics_QA",
+    "niddk_qa": "5_NIDDK_QA",
+    "ninds_qa": "6_NINDS_QA",
+    "seniorhealth_qa": "7_SeniorHealth_QA",
+    "nhlbi_qa_xml": "8_NHLBI_QA_XML",
+    "cdc_qa": "9_CDC_QA",
+    "mplus_adam_qa": "10_MPlus_ADAM_QA",
+    "mplusdrugs_qa": "11_MPlusDrugs_QA",
+    "mplusherbssupplements_qa": "12_MPlusHerbsSupplements_QA",
+}
+
+_URLS = {
+    "medquad_base_urls": _SUBSET_BASE_URIS,
+    f"QATestSetMedQrels_judged_answers": f"{_DATA_PATH}/QA-TestSet-LiveQA-Med-Qrels-2479-Answers.zip",
+}
+
+_SUPPORTED_TASKS = [Tasks.QUESTION_ANSWERING]  # TODO: shall we add a non-existing task type such as `RQE`?
```

Comment:
In the issue description, it says it supports QA and RQE, is it enough to put the _SUPPORTED_TASKS this way?

Reply:
Mmm... I think we should get away by using `constants.Tasks.TEXTUAL_ENTAILMENT`

Reply:
Quiting from the readme:
```
We included additional annotations in the XML files, that could be used for diverse IR and NLP tasks, such as the question type, the question focus, its syonyms, its UMLS Concept Unique Identifier (CUI) and Semantic Type.
```
So it seems it could be possible to add `NAMED_ENTITY_RECOGNTION/DISAMBIGUATION` to this but not sure...

Reply:
The data provided in the XML files doesn't seem to be structured as NER e.g. [this sample](https://github.com/abachaa/MedQuAD/blob/master/1_CancerGov_QA/0000004_3.xml)
I'll take another look to see if I could possibly parse them.